### ISSUE TITLE
Multi-cursor editing

### ISSUE THREAD

**Issue by notramo** [jmacdonald/amp#19]
The [Kakoune](https://kakoune.org/) editor is one of those that implements multi-cursor editing.
And it is very efficient, fast, and comfortable.
I think  *selection based editing* is very cool, and combining it with *multi-cursor editing* makes a very good experience.
Implementing these in `amp` needs a redesign, but I think it would worth the work.

**Comment by jmacdonald** [comment]
Neat, I've not heard of Kakoune!

I haven't integrated multiple cursors into my workflow, but I'd be happy to give it a shot to see if it would be a good fit, and how I could incorporate that into `amp`; let me look into that. 😄

**Comment by guillaumecherel** [comment]
Indeed, multi-cursor editing is great for refactoring. [Vis](https://github.com/martanne/vis) also does it well.

**Comment by notramo** [comment]
@jmacdonald Any progress?

**Comment by erf** [comment]
I think [vis](https://github.com/martanne/vis) has got this right by combining [sam's structural regular expressions](http://doc.cat-v.org/bell_labs/structural_regexps/) with multiple selections. This way you can do a search to 'collect' multiple selections and also do search on existing selections. ( there is also other way's to create those selections )

**Comment by notramo** [comment]
@erf Kakoune also has this feature: https://github.com/mawww/kakoune/issues/677

**Comment by erf** [comment]
@notramo Nice, i did not know that. I looked into Kakoune previously and found it a very nice editor except the keys are quite different than vim/vis.

**Comment by sjmackenzie** [comment]
Yes multi-cursor would be great to have.

**Comment by notramo** [comment]
@jmacdonald Any progress on this?
